src/main/java/com/ibm/cloud/sdk/core/http/RetryInterceptor.java

```diff
@@ -103,8 +133,13 @@ public Response intercept(Interceptor.Chain chain) throws IOException {
     return response;
   }
 
-  // Get the time we should wait before fire the next request in milliseconds.
-  private int getInterval(Response response, Request request) {
+  /**
+   * Determine the retry interval to wait before attempting the next retry.
+   * @param response the response from the previously attempted request
+   * @param request the previously attempted request
+   * @return the retry interval in milliseconds
+   */
+  protected int getInterval(Response response, Request request) {
```

Comment:
The primary methods in this class were changed from private to protected to enable users to subclass this implementation.  I think the primary use case for user-defined retry interceptors will probably be to subclass
this class and then override the "shouldRetry()" method below.